src/main/java/org/folio/dew/service/BulkEditParseService.java

```diff
@@ -0,0 +1,200 @@
+      return userGroup.getUsergroups().iterator().next().getId();
+    }
+    return null;
+  }
+
+  private List<UUID> getUserDepartments(UserFormat userFormat) {
+    String[] departmentNames = userFormat.getDepartments().split(ARRAY_DELIMITER);
+    if (departmentNames.length > 0) {
+      return Arrays.stream(departmentNames).parallel()
+        .filter(StringUtils::isNotEmpty)
+        .map(userReferenceService::getDepartmentByName)
+        .flatMap(departmentCollection -> departmentCollection.getDepartments().stream())
+        .map(Department::getId)
+        .map(UUID::fromString)
+        .collect(Collectors.toList());
+    }
+    return Collections.emptyList();
+  }
+
+  private List<String> getProxyFor(UserFormat userFormat) {
+    String[] proxyUserNames = userFormat.getProxyFor().split(ARRAY_DELIMITER);
+    if (proxyUserNames.length > 0) {
+      return Arrays.stream(proxyUserNames)
+        .parallel()
+        .filter(StringUtils::isNotEmpty)
+        .map(userReferenceService::getUserByName)
+        .flatMap(userCollection -> userCollection.getUsers().stream())
+        .map(User::getId)
+        .filter(StringUtils::isNotEmpty)
+        .map(userReferenceService::getProxyForByProxyUserId)
+        .flatMap(proxyForCollection -> proxyForCollection.getProxiesFor().stream())
+        .map(ProxyFor::getId)
+        .collect(Collectors.toList());
+    }
+    return Collections.emptyList();
+  }
+
+  private Personal getUserPersonalInfo(UserFormat userFormat) {
+    Personal personal = new Personal();
+    personal.setLastName(userFormat.getLastName());
+    personal.setFirstName(userFormat.getFirstName());
+    personal.setMiddleName(userFormat.getMiddleName());
+    personal.setPreferredFirstName(userFormat.getPreferredFirstName());
+    personal.setEmail(userFormat.getEmail());
+    personal.setPhone(userFormat.getPhone());
+    personal.setMobilePhone(userFormat.getMobilePhone());
+    personal.setDateOfBirth(getDate(userFormat.getDateOfBirth()));
+    personal.setAddresses(getUserAddresses(userFormat));
+    personal.setPreferredContactTypeId(userFormat.getPreferredContactTypeId());
+    return personal;
+  }
+
+  private Date getDate(String date) {
+    LocalDateTime localDateTime = LocalDateTime.parse(date, DateTimeFormatter.ofPattern(DATE_TIME_PATTERN));
+    return Date.from(localDateTime.atZone(UTC).toInstant());
+  }
+
+  private List<Address> getUserAddresses(UserFormat userFormat) {
+    String[] addresses = userFormat.getAddresses().split(ITEM_DELIMITER_PATTERN);
+    if (addresses.length > 0) {
+      return Arrays.stream(addresses)
+        .parallel()
+        .filter(StringUtils::isNotEmpty)
+        .map(this::getAddressFromString)
+        .collect(Collectors.toList());
+    }
+    return Collections.emptyList();
+  }
+
+  private Address getAddressFromString(String stringAddress) {
+    Address address = new Address();
```

Comment:
Check if stringAddress is not empty should be added

Reply:
The "stringAddress" cannot be empty here since the strings are filtered in a stream before being passed to this method.